Simplified/Book/UI/NYPLBookDetailDownloadFailedView.m

```diff
@@ -56,14 +60,23 @@ - (void)traitCollectionDidChange:(UITraitCollection *)previousTraitCollection {
     }
   }
 }
+- (void)setHidden:(BOOL)hidden {
+  [super setHidden:hidden];
+  /// The height of this view is constrainted to the normal view and downloading view.
+  /// If this is presenting an error message which takes more than 1 line,
+  /// we need to set the message to an empty string when this view is being hidden,
+  /// so the other views can resize to the height they need.
+  if (hidden) {
+    self.messageLabel.text = @"";
+  }
+}
 
 - (void)updateColors {
+  self.backgroundColor = [NYPLConfiguration mainColor];
```

Comment:
The original logic does not set the background color when it is iOS 12.0+ and in Light mode. This fixes it. Same for the one below.